contract/src/offer-up.contract.js

```diff
@@ -130,11 +132,42 @@ export const start = async zcf => {
    *   - want: `Items`
    */
   const makeTradeInvitation = () =>
-    zcf.makeInvitation(tradeHandler, 'buy items', undefined, proposalShape);
+    zcf.makeInvitation(
+      tradeHandler,
+      'buy subscription',
+      undefined,
+      proposalShape,
+    );
+
+  const isSubscriptionValid = userSubscription => {
+    if (!userSubscription || !userSubscription.value.payload) return false;
+
+    const serviceStarted = userSubscription.value.payload[0][0].serviceStarted;
+
+    // Here we'll check with current time from time service.
+    if (!serviceStarted || serviceStarted !== '123') return false;
+    return true;
+    //
+  };
+
+  const getSubscriptionResources = userAddress => {
+    const userSubscription = subscriptions.get(userAddress);
+
+    const isValidSub = isSubscriptionValid(userSubscription);
+    if (isValidSub) {
+      // User has a valid subscription, return the resources
+      const serviceType = userSubscription.value.payload[0][0].serviceType;
+      return subscriptionResources[serviceType];
+    } else {
+      // User doesn't have a valid subscription
+      return 'Access denied: You do not have a valid subscription.';
```

Comment:
@samsiegart what was the hackathon project you worked on where services would look for NFTs in wallets (using vstorage queries) to gate access?

Ah... right... https://github.com/agoric-labs/agoric-passport-express